### PR TITLE
MSR Door lock 

### PR DESCRIPTION
**Pull request by ThanatosGit**
Can be merged post or pre-stable. I don't care 😛

If post-stable:
- [x] Missing Changelog entry
- [x] Missing migration

### CODE REVIEW

**Comment by Miepee** [comment]
IMO should be post stable to ensure that it receives the proper testing

File: randovania/games/samus_returns/logic_database/header.txt [hunk 1]
```diff
@@ -434,7 +434,27 @@ Dock Weaknesses
...
+      Change from:
+          Charge Beam Door
+          Missile Door
+          Plasma Beam Door
+          Power Beam Door
+          Power Bomb Door
+          Spazer Beam Door
+          Super Missile Door
+          Wave Beam Door
+      Change to:
```
Comment (Miepee) [inline randovania/games/samus_returns/logic_database/header.txt:449]: not possible to change to access closed / permanently closed doors?

File: randovania/games/samus_returns/logic_database/Area 2 Dam Interior.txt [hunk 2]
```diff
@@ -834,7 +834,7 @@ Extra - polygon: [[-5800.0, -800.0], [-5800.0, -2800.0], [-4100.0, -2800.0], [-4
 Extra - asset_id: collision_camera041
 > Door to Wave Beam & Transport to Dam Exterior West; Heals? False
   * Layers: default
-  * Charge Beam Door to Wave Beam & Transport to Dam Exterior West/Door to Generator Access
+  * Charge Beam Door to Wave Beam & Transport to Dam Exterior West/Door to Generator Access; Dock Lock Rando incompatible with: Wave Beam Door
```
Comment (Miepee) [inline randovania/games/samus_returns/logic_database/Area 2 Dam Interior.txt:837]: would be nicer to get this change in seperately

File: test/test_files/log_files/samus_returns/vanilla_game.rdvgame [hunk 3]
```diff
@@ -1,622 +0,0 @@
-{
```
Comment (Miepee) [inline test/test_files/log_files/samus_returns/vanilla_game.rdvgame:1]: unsure if out of scope, but this rdvgame should not be completely deleted, but instead be used as resolver verification.

Comment (ThanatosGit) [reply test/test_files/log_files/samus_returns/vanilla_game.rdvgame:1]: It's unused. If anyone has the need to use it, they are free to add it again.

File: randovania/games/samus_returns/logic_database/header.txt (hunk 1 above)
Comment (ThanatosGit) [reply randovania/games/samus_returns/logic_database/header.txt:449]: Closed only exists with something else on the other side. A `doorclosedclosed` doesn't exist and have to be first implemented in the patcher.

**Comment by ThanatosGit** [comment]
> IMO should be post stable to ensure that it receives the proper testing

Could still be merged but excluded from stable. There are already several examples of stuff being merged but not pinged for beta test / ready for beta test.

**Comment by Miepee** [comment]
I'm blind, ignore me :sweat_smile:

**Comment by ThanatosGit** [comment]
Please stop force pushing over my branch creatinig conflicts between my local and remote especially with broken tests.
Make a proper merge and push it without a force.

**Comment by dyceron** [comment]
Wasn't intentional. Just a force off habit, my bad.

**Review by dyceron — APPROVED** [review]
Looks good to me, but I would rather have this merged post-stable to not entertain the thought of early playing of DLR.

**Review by Miepee — APPROVED** [review]

**Pull request merged**